components/activity_and_insights/insights/insights.tsx

```diff
@@ -1,24 +1,50 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-import React, {memo, useEffect} from 'react';
+import React, {memo, useState, useCallback, useEffect} from 'react';
 import {useDispatch} from 'react-redux';
 
 import {selectChannel} from 'mattermost-redux/actions/channels';
 
-import TopChannels from './top_channels/top_channels';
+import {localizeMessage} from 'utils/utils';
+
+import InsightsHeader from './insights_header/insights_header';
 
 import './../activity_and_insights.scss';
 
 const Insights = () => {
     const dispatch = useDispatch();
 
+    const [filterType, setFilterType] = useState('my');
+    const [timeFrame, setTimeFrame] = useState({
+        value: '1_day',
```

Comment:
Do you think it makes sense to have constants for `1_day` etc.?

Reply:
+1, these values will also be used for the modal state.

Reply:
Done